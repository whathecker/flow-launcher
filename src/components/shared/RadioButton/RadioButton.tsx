import React from "react";
import { Touchable, Text } from "../../Themed";
import { Container, Typography } from "../../../styles";

type RadioButtonProps = {
  pressHandler: () => void;
  value: string;
  displayText: string;
  activeValue: string;
};

const RadioButton: React.FC<RadioButtonProps> = ({
  displayText,
  value,
  activeValue,
  pressHandler,
}: RadioButtonProps) => {
  const isBtnActive = value === activeValue ? true : false;
  return (
    <Touchable
      onPress={pressHandler}
      lightColor="#ffffff"
      darkColor="#ffffff"
      style={{
        ...Container.centerAligned,
        width: "100%",
        height: 50,
        paddingTop: 10,
        paddingBottom: 10,
        marginTop: 10,
        marginBottom: 10,
        marginLeft: 3,
        marginRight: 3,
        borderRadius: 10,
        borderWidth: isBtnActive ? 2 : 1,
        borderColor: isBtnActive ? "#0FA858" : "#848484",
        borderStyle: "solid",
        overflow: "hidden",
      }}
    >
      <Text
        lightColor={isBtnActive ? "#0FA858" : "#848484"}
        darkColor={isBtnActive ? "#0FA858" : "#848484"}
        style={{ ...Typography.p, fontSize: 15 }}
      >
        {displayText}
      </Text>
    </Touchable>
  );
};

export default RadioButton;
