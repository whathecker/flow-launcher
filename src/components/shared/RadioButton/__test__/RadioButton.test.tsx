import * as React from "react";
import renderer from "react-test-renderer";
import RadioButton from "../RadioButton";

describe("Test RadioButton component", () => {
  it("Should renders inactive RadioButton correctly", () => {
    const tree = renderer
      .create(
        <RadioButton
          value="daily"
          displayText="Every Day"
          activeValue="three_days"
          pressHandler={() => {
            return;
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should renders active RadioButton correctly", () => {
    const tree = renderer
      .create(
        <RadioButton
          value="seven_days"
          displayText="Every Day"
          activeValue="seven_days"
          pressHandler={() => {
            return;
          }}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
