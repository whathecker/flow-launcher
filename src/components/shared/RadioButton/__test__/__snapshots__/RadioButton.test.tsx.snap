// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RadioButton component Should renders active RadioButton correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#0FA858",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 3,
      "marginRight": 3,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#0FA858",
        },
        Object {
          "fontFamily": "Lexend-Regular",
          "fontSize": 15,
        },
      ]
    }
  >
    Every Day
  </Text>
</View>
`;

exports[`Test RadioButton component Should renders inactive RadioButton correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#848484",
      "borderRadius": 10,
      "borderStyle": "solid",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 3,
      "marginRight": 3,
      "marginTop": 10,
      "opacity": 1,
      "overflow": "hidden",
      "paddingBottom": 10,
      "paddingTop": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#848484",
        },
        Object {
          "fontFamily": "Lexend-Regular",
          "fontSize": 15,
        },
      ]
    }
  >
    Every Day
  </Text>
</View>
`;
